ui/src/app/components/with-workspace-gke-apps.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+
+import { UserAppEnvironment } from 'generated/fetch';
+
+import { Spinner } from 'app/components/spinners';
+import { appsApi } from 'app/services/swagger-fetch-clients';
+import { notificationStore } from 'app/utils/stores';
+
+export interface WithWorkspaceGKEAppsProps {
+  workspaceNamespace: string;
+}
+
+interface WrappedComponentRequiredProps {
+  gkeAppsInWorkspace: NonNullable<UserAppEnvironment[]>;
+}
+
+// Given a Component, returns a new Component that injects a gkeAppsInWorkspace prop into the original component.
+// TODO: Validate that gkeAppsInWorkspace in WrappedComponent is the correct type
```

Comment:
I spent awhile trying to ensure that the WrappedComponent's props contained a `NonNullable<UserAppEnvironment[]>` named `gkeAppsInWorkspace`. I think it's fine as-is and I was mostly nerd-sniping myself. This TODO is intentional in case we want to revisit it or if a future Typescript version addresses it but I'm open to removing the TODO if it's not useful.